sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -69,6 +69,7 @@ trait BaseLimitExec extends UnaryExecNode with CodegenSupport {
   override def doConsume(ctx: CodegenContext, input: Seq[ExprCode], row: ExprCode): String = {
     val stopEarly = ctx.freshName("stopEarly")
     ctx.addMutableState("boolean", stopEarly, s"$stopEarly = false;")
+    shouldStopRequired = true // loop may break early even without append in loop body
```

Comment:
If this `Limit` is the parent of an `Aggregate`, the final `shouldStopRequired` is true. But actually we can skip the check.